packages/local_auth/local_auth_darwin/darwin/local_auth_darwin/Sources/local_auth_darwin/LocalAuthPlugin.swift

```diff
@@ -0,0 +1,421 @@
+class DefaultViewProvider: ViewProvider {
+  /// The wrapped registrar.
+  let registrar: FlutterPluginRegistrar
+
+  /// Returns a wrapper for the given FlutterPluginRegistrar.
+  init(registrar: FlutterPluginRegistrar) {
+    self.registrar = registrar
+  }
+
+  #if os(macOS)
+    var view: NSView? {
+      return registrar.view
+    }
+  #endif  // os(macOS)
+}
+
+// MARK: -
+
+/// A data container for sticky auth state.
+struct StickyAuthState {
+  let options: AuthOptions
+  let strings: AuthStrings
+  let resultHandler: (Result<AuthResultDetails, Error>) -> Void
+}
+
+// MARK: -
+
+/// A flutter plugin for local authentication.
+// TODO(stuartmorgan): Remove the @unchecked Sendable, and convert to strict thread enforcement.
+// This was added to minimize the changes while converting from Swift to Objective-C.
+public final class LocalAuthPlugin: NSObject, FlutterPlugin, LocalAuthApi, @unchecked Sendable {
+
+  /// The factory to create LAContexts.
+  private let authContextFactory: AuthContextFactory
+  /// The factory to create alerts.
+  private let alertFactory: AuthAlertFactory
+  /// The Flutter view provider.
+  private let viewProvider: ViewProvider
+  /// Manages the last call state for sticky auth.
+  private var lastCallState: StickyAuthState?
+
+  public static func register(with registrar: FlutterPluginRegistrar) {
+    let instance = LocalAuthPlugin(
+      contextFactory: DefaultAuthContextFactory(),
+      alertFactory: DefaultAlertFactory(),
+      viewProvider: DefaultViewProvider(registrar: registrar))
+    registrar.addApplicationDelegate(instance)
+    // Workaround for https://github.com/flutter/flutter/issues/118103.
+    #if os(iOS)
+      let messenger = registrar.messenger()
+    #else
+      let messenger = registrar.messenger
+    #endif
+    LocalAuthApiSetup.setUp(binaryMessenger: messenger, api: instance)
+  }
+
+  /// Returns an instance that uses the given factory to create LAContexts.
+  init(
+    contextFactory: AuthContextFactory,
+    alertFactory: AuthAlertFactory,
+    viewProvider: ViewProvider
+  ) {
+    self.authContextFactory = contextFactory
+    self.alertFactory = alertFactory
+    self.viewProvider = viewProvider
+  }
+
+  // MARK: LocalAuthApi
+
+  func authenticate(
+    options: AuthOptions,
+    strings: AuthStrings,
+    completion: @escaping (Result<AuthResultDetails, Error>) -> Void
+  ) {
+    var context = authContextFactory.createAuthContext()
+    lastCallState = nil
+    context.localizedFallbackTitle = strings.localizedFallbackTitle
+
+    let policy =
+      options.biometricOnly
+      ? LAPolicy.deviceOwnerAuthenticationWithBiometrics
+      : LAPolicy.deviceOwnerAuthentication
+    var authError: NSError?
+    if context.canEvaluatePolicy(policy, error: &authError) {
+      context.evaluatePolicy(
+        policy,
+        localizedReason: strings.reason
+      ) { (success: Bool, error: Error?) in
+        DispatchQueue.main.async { [weak self] in
```

Comment:
I think you have to put [weak self] in outer closure `context.evaluatePolicy`, otherwise it would remain strong when outer closure is still alive.

Reply:
Makes sense; done.